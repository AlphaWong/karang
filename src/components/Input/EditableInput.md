#### Usage:

```js static
import { EditableInput } from '@lalamove/karang';
```

```js
initialState = { email: 'no-reply@lalamove.com' };
<EditableInput
    label="Email"
    value={state.email} 
    onSave={value => setState({ email: value })} 
/>
```
