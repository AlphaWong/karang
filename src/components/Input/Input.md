#### Usage:

```js static
import { Input } from 'lalamove-ui';
```

```js
initialState = { name: 'Alex Fok' };
<Input label="Name" value={state.name} onChange={e => setState({ name: e.target.value })}/>
```
