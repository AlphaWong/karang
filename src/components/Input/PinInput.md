#### Usage:

```js static
import { PinInput } from 'lalamove-ui';
```

```js

```
