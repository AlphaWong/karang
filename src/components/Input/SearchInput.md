#### Usage:

```js static
import { SearchInput } from 'lalamove-ui';
```

```js
<SearchInput placeholder="Search for Order ID, Driver's Mobile" />
```
