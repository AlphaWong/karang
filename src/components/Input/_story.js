import React from 'react';
import { storiesOf } from '@storybook/react';
import AnimatedInput from './index';
import withAutoFocus from '../../hoc/withAutoFocus';
import withOnClickSelect from '../../hoc/withOnClickSelect';
import withOnClickToEnd from '../../hoc/withOnClickToEnd';

const InputWithAutoFocus = withAutoFocus(AnimatedInput, 'innerRef');
const InputWithOnClickSelect = withOnClickSelect(AnimatedInput, 'innerRef');
const InputWithOnClickToEnd = withOnClickToEnd(AnimatedInput, 'innerRef');

// TODO: Clear it up & add withInfo
storiesOf('Input', module).add('Basic', () => (
  <div style={{ padding: '20px' }}>
    <AnimatedInput
      type="text"
      placeholder="Input"
      name="username"
      value=""
      autoComplete="off"
    />
    <br />
    <AnimatedInput
      type="password"
      placeholder="Password"
      name="password"
      value=""
    />
    <br />
    <InputWithAutoFocus
      type="text"
      placeholder="InputWithAutoFocus"
      name="inputWithAutoFocus"
      value=""
      autoComplete="off"
    />
    <br />
    <InputWithOnClickSelect
      type="text"
      placeholder="InputWithOnClickSelect"
      name="inputWithOnClickSelect"
      value=""
      autoComplete="off"
    />
    <br />
    <InputWithOnClickToEnd
      type="text"
      placeholder="InputWithOnClickToEnd"
      name="inputWithOnClickToEnd"
      value=""
      autoComplete="off"
    />
  </div>
));
