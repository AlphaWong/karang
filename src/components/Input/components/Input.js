import React, { Component, Fragment, forwardRef } from 'react';
import { bool, func, string, object, oneOfType } from 'prop-types';
import { branch, compose, toClass } from 'recompose';
import styled from 'styled-components';

import withAnimatedContainer from 'hoc/withAnimatedContainer';
import withErrorMessage from 'hoc/withErrorMessage';
import withSelectAll from 'hoc/withSelectAll';
import withCursorEnd from 'hoc/withCursorEnd';

import TextInput from './TextInput';
import PeekButton from './PeekButton';

const StyledTextInput = styled(TextInput)`
  &::-ms-reveal {
    display: none;
  }
`;

class Input extends Component {
  static propTypes = {
    innerRef: oneOfType([func, object]),
    type: string,
    name: string.isRequired,
    label: string,
    placeholder: string,
    error: string,
    autoComplete: string,
    selectAll: bool,
    cursorEnd: bool,
  };

  static defaultProps = {
    innerRef: null,
    type: 'text',
    label: '',
    placeholder: '',
    error: null,
    // autocomplete=off is ignored on non-login INPUT elements
    // https://bugs.chromium.org/p/chromium/issues/detail?id=468153#c164
    autoComplete: 'new-password',
    selectAll: false,
    cursorEnd: false,
  };

  state = {
    peekPassword: false,
  };

  changePeekStatus = () => {
    this.setState(prevState => ({
      peekPassword: !prevState.peekPassword,
    }));
  };

  render() {
    const { peekPassword } = this.state;
    const {
      type,
      name,
      label,
      placeholder,
      autoComplete,
      error,
      selectAll,
      cursorEnd,
      innerRef,
      ...remainProps
    } = this.props;

    return (
      <Fragment>
        <StyledTextInput
          type={peekPassword ? 'text' : type}
          name={name}
          label={label}
          placeholder={placeholder}
          autoComplete={autoComplete}
          {...remainProps}
          innerRef={innerRef}
        />
        {type === 'password' && (
          <PeekButton active={peekPassword} onClick={this.changePeekStatus} />
        )}
      </Fragment>
    );
  }
}

const InputWithRef = forwardRef((props, ref) => (
  <Input {...props} innerRef={ref} />
));

const EnhancedComp = compose(
  branch(props => props.selectAll, withSelectAll),
  toClass,
  branch(props => props.cursorEnd, withCursorEnd),
  withErrorMessage,
  withAnimatedContainer
)(InputWithRef);

export default EnhancedComp;
