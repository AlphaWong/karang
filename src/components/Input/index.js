import Input from './components/Input';
import PinInput from './components/PinInput';
import SearchInput from './components/SearchInput';
import EditableInput from './components/EditableInput';

export { Input as default, PinInput, SearchInput, EditableInput };
