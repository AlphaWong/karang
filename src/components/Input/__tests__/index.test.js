import React from 'react';
import { render } from 'enzyme';
import Input from '../index';

// Enzyme is still not fully support React 16.3 forwardRef
// https://github.com/airbnb/enzyme/pull/1592
// https://github.com/airbnb/enzyme/issues/1717

describe('Input', () => {
  describe('Snapshots', () => {
    it('Password', () => {
      const wrapper = render(<Input name="input" type="password" />);
      expect(wrapper).toMatchSnapshot();
    });

    // it('Password with a value', () => {
    //   const wrapper = mount(
    //     <Input name="input" type="password" value="dummy" />
    //   );
    //   expect(wrapper).toMatchSnapshot();
    // });

    it('Regular Input', () => {
      const wrapper = render(<Input name="input" />);
      expect(wrapper).toMatchSnapshot();
    });
  });

  // describe('Functions', () => {
  //   it('trigger onClick when clicked', () => {
  //     const clickHandler = jest.fn();
  //     const wrapper = mount(<Input name="input" onClick={clickHandler} />);
  //     wrapper.find('input').simulate('click');
  //     expect(clickHandler).toHaveBeenCalledTimes(1);
  //   });
  //
  //   it('trigger onChange when changed', () => {
  //     const changeHandler = jest.fn();
  //     const wrapper = mount(<Input name="input" onChange={changeHandler} />);
  //     wrapper.find('input').simulate('change');
  //     expect(changeHandler).toHaveBeenCalledTimes(1);
  //   });
  //
  //   it('trigger onBlur when blurred', () => {
  //     const blurHandler = jest.fn();
  //     const wrapper = mount(<Input name="input" onBlur={blurHandler} />);
  //     wrapper.find('input').simulate('blur');
  //     expect(blurHandler).toHaveBeenCalledTimes(1);
  //   });
  //
  //   it('trigger onFocus when focused', () => {
  //     const focusHandler = jest.fn();
  //     const wrapper = mount(<Input name="input" onFocus={focusHandler} />);
  //     wrapper.find('input').simulate('focus');
  //     expect(focusHandler).toHaveBeenCalledTimes(1);
  //   });
  // });
  //
  // describe('Password', () => {
  //   it('contains the PeekButton', () => {
  //     const wrapper = shallow(<Input name="input" type="password" />);
  //     expect(wrapper.find('PeekButton').exists()).toBe(true);
  //   });
  // });
  //
  // describe('Input', () => {
  //   it('check the input value based on value prop', () => {
  //     const wrapper = mount(<Input name="input" value="test" />);
  //     expect(wrapper.find('input').instance().value).toEqual('test');
  //   });
  //
  //   it('check input value after change', () => {
  //     const wrapper = mount(<Input name="input" />);
  //     wrapper.find('input').simulate('change', { target: { value: 'test' } });
  //     expect(wrapper.find('input').instance().value).toEqual('test');
  //   });
  // });
  //
  // describe('Placeholder', () => {
  //   it('check the placeholder value based on placeholder prop', () => {
  //     const wrapper = mount(<Input name="input" placeholder="test" />);
  //     expect(wrapper.find('Placeholder').text()).toEqual('test');
  //   });
  // });
  //
  // describe('Error Message', () => {
  //   it('check if the error message is rendered subject to an existing prop', () => {
  //     const wrapper = shallow(<Input name="input" error="Test Error" />);
  //     expect(wrapper.find('ErrorMessage').exists()).toBe(true);
  //   });
  // });
});

// describe('PinInput', () => {
//   let props;
//   let mountedPinInput;
//   const pinInput = () => {
//     if (!mountedPinInput) {
//       mountedPinInput = mount(<PinInput {...props} />);
//     }
//     return mountedPinInput;
//   };
//
//   beforeEach(() => {
//     props = {
//       pins: undefined,
//       disabled: undefined,
//       error: undefined,
//       onChange: undefined,
//     };
//     mountedPinInput = undefined;
//   });
//
//   it('always renders a div', () => {
//     const divs = pinInput().find('div');
//     expect(divs.length).toBeGreaterThan(0);
//   });
//
//   describe('the rendered div', () => {
//     it('contains everything else that gets rendered', () => {
//       const divs = pinInput().find('div');
//       const wrappingDiv = divs.first();
//       expect(wrappingDiv).toEqual(pinInput().children());
//     });
//   });
//
//   it('always renders 4 inputs', () => {
//     expect(pinInput().find('input').length).toBe(4);
//   });
//
//   describe('when error is passed', () => {
//     beforeEach(() => {
//       props.error = 'Error message.';
//     });
//
//     it('renders an error div', () => {
//       expect(pinInput().find('div.PinInput__error').length).toBe(1);
//     });
//
//     it('passes error message to the rendered error div', () => {
//       const errorDiv = pinInput().find('div.PinInput__error');
//       expect(errorDiv.childAt(0).props().children).toBe(props.error);
//     });
//   });
//
//   describe('when error is undefined', () => {
//     beforeEach(() => {
//       props.error = undefined;
//     });
//
//     it('does not render any error div', () => {
//       expect(pinInput().find('div.PinInput__error').length).toBe(0);
//     });
//   });
// });
