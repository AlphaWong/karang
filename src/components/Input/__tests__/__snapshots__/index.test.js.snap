// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Snapshots Password 1`] = `
.c2 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
  border: 1px solid rgba(229,229,229,1);
  font-family: Noto Sans,"Noto Sans TC","Noto Sans JP",sans-serif;
  text-align: left;
}

.c0 {
  display: inline-block;
}

.c4 {
  width: 100%;
  padding: 1em;
  border: none;
  color: rgba(51,49,50,1);
  font-family: Noto Sans,"Noto Sans TC","Noto Sans JP",sans-serif;
  caret-color: rgba(242,103,34,1);
  font-size: 14px;
  line-height: 20px;
  resize: none;
  outline: none;
}

.c5 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: none;
  background: none;
  cursor: pointer;
  margin: 0 14px;
  outline: 0;
  padding: 0;
}

.c3::-ms-reveal {
  display: none;
}

.c1 {
  width: 100%;
}

<div
  class="c0"
>
  <div
    class="c1 c2"
  >
    <input
      autocomplete="new-password"
      class="c3 c4"
      id="llmInput_input"
      label=""
      name="input"
      placeholder=""
      type="password"
    >
      
    </input>
    <button
      class="c5"
    >
      <svg
        style="vertical-align:middle;fill:rgba(51,49,50,1);width:20px;height:20px"
        version="1.1"
        viewBox="0 0 1024 1024"
        xlink="http://www.w3.org/1999/xlink"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g>
          <path
            d="M818.728 176.959c7.569 0 14.676 2.949 20.018 8.296 5.351 5.347 8.296 12.448 8.296 20.018s-2.945 14.671-8.296 20.018l-613.452 613.452c-5.347 5.351-12.453 8.301-20.022 8.301s-14.676-2.949-20.018-8.296c-5.351-5.347-8.296-12.448-8.296-20.018s2.945-14.671 8.296-20.018l613.452-613.452c5.347-5.351 12.453-8.301 20.022-8.301zM512 623.959c-5.828 0-11.585-0.448-17.233-1.293l127.896-127.906c0.854 5.649 1.302 11.406 1.302 17.238 0 30.149-11.585 58.33-32.612 79.358-21.023 21.023-49.204 32.603-79.353 32.603zM512 400.041c5.833 0 11.59 0.448 17.233 1.293l-127.891 127.901c-0.854-5.649-1.298-11.401-1.298-17.233 0-30.149 11.58-58.33 32.608-79.358 21.023-21.023 49.204-32.603 79.348-32.603zM933.862 507.272l2.841 4.728-2.841 4.728c-39.832 66.442-101.234 126.466-172.886 169.012-77.489 46.014-163.58 70.34-248.974 70.34-44.141 0-88.408-6.522-131.553-19.102l47.571-47.571c25.477 12.109 53.946 18.909 83.982 18.909 108.251 0 196.32-88.069 196.32-196.315 0-30.036-6.8-58.505-18.913-83.982l35.217-35.217c20.381 35.991 31.451 76.923 31.451 119.199 0 54.956-18.64 107.676-52.233 150.174 97.705-49.662 157.224-124.555 175.543-150.018-46.67-61.898-97.035-103.584-142.473-131.638l35.354-35.359c66.777 42.031 123.88 99.201 161.594 162.113zM144.502 512c16.030 22.306 64.248 83.317 142.52 131.549l-35.269 35.278c-66.72-42.022-123.833-99.191-161.608-162.089l-2.845-4.738 2.841-4.738c39.823-66.305 101.074-126.235 172.48-168.757 77.546-46.174 163.774-70.585 249.379-70.585 44.126 0 88.394 6.522 131.553 19.102l-47.571 47.566c-25.477-12.104-53.946-18.909-83.982-18.909-108.251 0-196.315 88.073-196.315 196.32 0 30.036 6.805 58.5 18.909 83.982l-35.217 35.217c-20.381-35.991-31.451-76.923-31.451-119.199 0-54.956 18.644-107.676 52.238-150.174-97.903 49.765-157.474 124.876-175.661 150.174z"
          >
            
          </path>
        </g>
      </svg>
    </button>
  </div>
</div>
`;

exports[`Input Snapshots Password with a value 1`] = `
.c2 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
  border: 1px solid rgba(229,229,229,1);
  font-family: Noto Sans,"Noto Sans TC","Noto Sans JP",sans-serif;
  text-align: left;
}

.c0 {
  display: inline-block;
}

.c4 {
  width: 100%;
  padding: 1em;
  border: none;
  color: rgba(51,49,50,1);
  font-family: Noto Sans,"Noto Sans TC","Noto Sans JP",sans-serif;
  caret-color: rgba(242,103,34,1);
  font-size: 14px;
  line-height: 20px;
  resize: none;
  outline: none;
}

.c5 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: none;
  background: none;
  cursor: pointer;
  margin: 0 14px;
  outline: 0;
  padding: 0;
}

.c3::-ms-reveal {
  display: none;
}

.c1 {
  width: 100%;
}

<div
  class="c0"
>
  <div
    class="c1 c2"
  >
    <input
      autocomplete="new-password"
      class="c3 c4"
      id="llmInput_input"
      label=""
      name="input"
      placeholder=""
      type="password"
      value="dummy"
    >
      
    </input>
    <button
      class="c5"
    >
      <svg
        style="vertical-align:middle;fill:rgba(51,49,50,1);width:20px;height:20px"
        version="1.1"
        viewBox="0 0 1024 1024"
        xlink="http://www.w3.org/1999/xlink"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g>
          <path
            d="M818.728 176.959c7.569 0 14.676 2.949 20.018 8.296 5.351 5.347 8.296 12.448 8.296 20.018s-2.945 14.671-8.296 20.018l-613.452 613.452c-5.347 5.351-12.453 8.301-20.022 8.301s-14.676-2.949-20.018-8.296c-5.351-5.347-8.296-12.448-8.296-20.018s2.945-14.671 8.296-20.018l613.452-613.452c5.347-5.351 12.453-8.301 20.022-8.301zM512 623.959c-5.828 0-11.585-0.448-17.233-1.293l127.896-127.906c0.854 5.649 1.302 11.406 1.302 17.238 0 30.149-11.585 58.33-32.612 79.358-21.023 21.023-49.204 32.603-79.353 32.603zM512 400.041c5.833 0 11.59 0.448 17.233 1.293l-127.891 127.901c-0.854-5.649-1.298-11.401-1.298-17.233 0-30.149 11.58-58.33 32.608-79.358 21.023-21.023 49.204-32.603 79.348-32.603zM933.862 507.272l2.841 4.728-2.841 4.728c-39.832 66.442-101.234 126.466-172.886 169.012-77.489 46.014-163.58 70.34-248.974 70.34-44.141 0-88.408-6.522-131.553-19.102l47.571-47.571c25.477 12.109 53.946 18.909 83.982 18.909 108.251 0 196.32-88.069 196.32-196.315 0-30.036-6.8-58.505-18.913-83.982l35.217-35.217c20.381 35.991 31.451 76.923 31.451 119.199 0 54.956-18.64 107.676-52.233 150.174 97.705-49.662 157.224-124.555 175.543-150.018-46.67-61.898-97.035-103.584-142.473-131.638l35.354-35.359c66.777 42.031 123.88 99.201 161.594 162.113zM144.502 512c16.030 22.306 64.248 83.317 142.52 131.549l-35.269 35.278c-66.72-42.022-123.833-99.191-161.608-162.089l-2.845-4.738 2.841-4.738c39.823-66.305 101.074-126.235 172.48-168.757 77.546-46.174 163.774-70.585 249.379-70.585 44.126 0 88.394 6.522 131.553 19.102l-47.571 47.566c-25.477-12.104-53.946-18.909-83.982-18.909-108.251 0-196.315 88.073-196.315 196.32 0 30.036 6.805 58.5 18.909 83.982l-35.217 35.217c-20.381-35.991-31.451-76.923-31.451-119.199 0-54.956 18.644-107.676 52.238-150.174-97.903 49.765-157.474 124.876-175.661 150.174z"
          >
            
          </path>
        </g>
      </svg>
    </button>
  </div>
</div>
`;

exports[`Input Snapshots Regular Input 1`] = `
.c2 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
  border: 1px solid rgba(229,229,229,1);
  font-family: Noto Sans,"Noto Sans TC","Noto Sans JP",sans-serif;
  text-align: left;
}

.c0 {
  display: inline-block;
}

.c4 {
  width: 100%;
  padding: 1em;
  border: none;
  color: rgba(51,49,50,1);
  font-family: Noto Sans,"Noto Sans TC","Noto Sans JP",sans-serif;
  caret-color: rgba(242,103,34,1);
  font-size: 14px;
  line-height: 20px;
  resize: none;
  outline: none;
}

.c3::-ms-reveal {
  display: none;
}

.c1 {
  width: 100%;
}

<div
  class="c0"
>
  <div
    class="c1 c2"
  >
    <input
      autocomplete="new-password"
      class="c3 c4"
      id="llmInput_input"
      label=""
      name="input"
      placeholder=""
      type="text"
    >
      
    </input>
  </div>
</div>
`;

exports[`PinInput Snapshots 4 digits filled with error message 1`] = `
.c1 {
  display: inline-block;
}

.c3 {
  margin-top: 0.3em;
  margin-bottom: 0.3em;
  line-height: 1.4;
}

.c5 {
  color: rgba(216,83,79,1);
  font-family: Noto Sans,"Noto Sans TC","Noto Sans JP",sans-serif;
  font-size: 12px;
}

.c4 {
  float: right;
  margin-left: 1em;
}

.c0 {
  display: inline-block;
}

.c2 {
  border: 1px solid rgba(229,229,229,1);
  caret-color: rgba(242,103,34,1);
  font-family: Noto Sans,"Noto Sans TC","Noto Sans JP",sans-serif;
  width: 68px;
  height: 68px;
  font-size: 48px;
  outline: 0;
  margin: 0 8px 8px 8px;
  text-align: center;
  border: 1px solid rgba(216,83,79,1);
  color: rgba(216,83,79,1);
}

.c2:focus {
  border: 1px solid rgba(242,103,34,1);
}

.c2:focus {
  border: 1px solid rgba(216,83,79,1);
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div>
      <input
        autocomplete="new-password"
        autofocus=""
        class="c2"
        maxlength="1"
        name="0"
        value="1"
      >
        
      </input>
      <input
        autocomplete="new-password"
        class="c2"
        maxlength="1"
        name="1"
        value="2"
      >
        
      </input>
      <input
        autocomplete="new-password"
        class="c2"
        maxlength="1"
        name="2"
        value="3"
      >
        
      </input>
      <input
        autocomplete="new-password"
        class="c2"
        maxlength="1"
        name="3"
        value="4"
      >
        
      </input>
    </div>
    <div
      class="c3"
    >
      <span
        class="c4"
      >
        <svg
          style="vertical-align:middle;fill:rgba(216,83,79,1);width:13px;height:13px"
          version="1.1"
          viewBox="0 0 1024 1024"
          xlink="http://www.w3.org/1999/xlink"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <path
              d="M512.005 132.673c-22.094 0-44.183 8.419-61.039 25.265-6.116 6.106-11.141 12.949-14.949 20.187v-0.052l-333.777 578.546c-5.271 7.734-9.457 16.469-11.972 26.072-12.349 46.080 14.973 93.448 61.020 105.727 7.385 1.987 14.855 2.912 22.254 2.912 0.892 0 1.784-0.014 2.676-0.042l667.851-0.184c2.010 0.137 4.030 0.208 6.031 0.208 29.913 0 58.944-15.36 75.030-43.065 16.393-28.507 14.836-62.45-1.326-88.526l-334.126-578.23c-4.186-8.503-9.584-16.483-16.639-23.552-16.851-16.846-38.94-25.265-61.034-25.265zM144.71 756.496v0zM512.005 175.143c11.717 0 22.731 4.558 30.965 12.783 3.374 3.388 6.191 7.418 8.607 12.316 0.897 1.831 1.925 3.582 3.067 5.247l332.385 575.469c0.222 0.382 0.444 0.76 0.675 1.133 8.47 13.671 8.706 30.904 0.698 44.829-7.97 13.723-22.287 21.919-38.303 21.919-1.024 0-2.067-0.033-3.096-0.109-0.972-0.066-1.944-0.099-2.916-0.099-0.821 0-1.642 0.024-2.449 0.071l-665.425 0.113c-0.42 0-0.845 0.009-1.265 0.019-0.467 0.019-0.934 0.024-1.406 0.024-3.865 0-7.649-0.491-11.311-1.477-11.273-3.006-20.702-10.235-26.553-20.348-5.87-10.164-7.428-22.004-4.389-33.348 0.019-0.076 0.042-0.156 0.061-0.236 1.17-4.455 3.185-8.801 5.988-12.911 1-1.477 1.911-3.015 2.718-4.601l331.96-575.276c0.566-0.901 1.1-1.826 1.595-2.779 1.873-3.553 4.346-6.885 7.376-9.905 8.282-8.277 19.3-12.835 31.017-12.835zM512 793.378c23.458 0 42.47-19.012 42.47-42.47s-19.012-42.47-42.47-42.47c-23.458 0-42.47 19.012-42.47 42.47s19.012 42.47 42.47 42.47zM554.47 622.889v-255.42c0-23.458-19.012-42.47-42.47-42.47s-42.47 19.012-42.47 42.47v255.42c0 23.458 19.012 42.47 42.47 42.47s42.47-19.012 42.47-42.47z"
            >
              
            </path>
          </g>
        </svg>
      </span>
      <span
        class="c5"
      >
        Error Message
      </span>
    </div>
  </div>
</div>
`;

exports[`PinInput Snapshots Empty 1`] = `
.c1 {
  display: inline-block;
}

.c0 {
  display: inline-block;
}

.c2 {
  border: 1px solid rgba(229,229,229,1);
  caret-color: rgba(242,103,34,1);
  font-family: Noto Sans,"Noto Sans TC","Noto Sans JP",sans-serif;
  width: 68px;
  height: 68px;
  font-size: 48px;
  outline: 0;
  margin: 0 8px 8px 8px;
  text-align: center;
}

.c2:focus {
  border: 1px solid rgba(242,103,34,1);
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div>
      <input
        autocomplete="new-password"
        autofocus=""
        class="c2"
        maxlength="1"
        name="0"
        value=""
      >
        
      </input>
      <input
        autocomplete="new-password"
        class="c2"
        maxlength="1"
        name="1"
        value=""
      >
        
      </input>
      <input
        autocomplete="new-password"
        class="c2"
        maxlength="1"
        name="2"
        value=""
      >
        
      </input>
      <input
        autocomplete="new-password"
        class="c2"
        maxlength="1"
        name="3"
        value=""
      >
        
      </input>
    </div>
  </div>
</div>
`;
